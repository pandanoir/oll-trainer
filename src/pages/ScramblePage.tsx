import { useEffect, useState } from 'react';
import SwiperCore, { Navigation, Keyboard } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
import Scrambo from 'scrambo';

SwiperCore.use([Navigation, Keyboard]);

export const ScramblePage = () => {
  const [scrambles, setScrambles] = useState<string[]>([]);
  const [index, setIndex] = useState(0);
  useEffect(() => {
    if (index + 3 >= scrambles.length) {
      const id = setTimeout(() => {
        setScrambles(scrambles.concat(new Scrambo().get(3)));
      }, 1000 / 60);
      return () => clearTimeout(id);
    }
  });
  return (
    <div className="w-screen">
      <div className="font-bold text-3xl text-center">#{index + 1}</div>
      <Swiper
        onSlideChange={({ activeIndex }: { activeIndex: number }) =>
          setIndex(activeIndex)
        }
        keyboard
        spaceBetween={50}
        navigation
      >
        {scrambles.map((scramble, index) => (
          <SwiperSlide key={index}>{scramble}</SwiperSlide>
        ))}
      </Swiper>
    </div>
  );
};
