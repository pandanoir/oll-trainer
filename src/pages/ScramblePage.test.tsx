/**
 * @jest-environment jsdom
 */
import { act, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ScramblePage } from './ScramblePage';

jest.useFakeTimers();
describe('ScramblePage', () => {
  beforeEach(() => {
    const randomValues = [
      0.2617694612221786, 0.4155313799639526, 0.4278997503749007,
      0.673774560722304, 0.18678664330210504, 0.7388180785604643,
      0.08093411023266106, 0.6852704318998923, 0.822943378927079,
      0.4207375154160369, 0.8737941266330909, 0.5175097214493198,
      0.5111515340713797, 0.012568834046551958, 0.5846614931966014,
      0.7181151068325067, 0.6905858007670413, 0.7513769600889282,
      0.2161180115140009, 0.09289620851961389, 0.32123058412926664,
      0.07676367011080298, 0.7730790993414411, 0.5528692434435669,
      0.7104639995387054, 0.47329369245923525, 0.5355846014959236,
      0.845933694732028, 0.8743663496668377, 0.525336444911018,
      0.13958888970636063, 0.5711988471342653, 0.8181816790930492,
      0.22229069268797574, 0.33755800126180824, 0.5681123733188622,
      0.16634332275298624, 0.27222967068198156, 0.16141417923774637,
      0.5082656252045381, 0.37729922279254247, 0.3359912282483646,
      0.5127349746611132, 0.9779093484852612, 0.7158443696597636,
      0.8265854858380688, 0.5319480807357104, 0.1400465333947516,
      0.5325290130643807, 0.7147377866506262, 0.039155150146721995,
      0.36679874411817015, 0.8909059408812621, 0.6801751963417237,
      0.023484146249223903, 0.2662478593201554, 0.138629720238459,
      0.1035829176284313, 0.8282662545971753, 0.17136083791231438,
      0.6040504451166215, 0.9273740481976, 0.005252297304446429,
      0.24045846692489126, 0.29373114924136523, 0.2059002230135043,
      0.6007470964953427, 0.8870845206124705, 0.6538746197349619,
      0.42054118954387865, 0.0014582632924344807, 0.4827710466375954,
      0.010728892620013308, 0.7988503825191284, 0.7378285466045775,
      0.2624489307868596, 0.5705825601985712, 0.028529624665165843,
      0.0815274126411285, 0.7658616116328061, 0.05381296439689209,
      0.8228355773827669, 0.5601672306739023, 0.17438560044481255,
      0.9671990072122085, 0.2734900253086523, 0.0741747186483368,
      0.04356794052184454, 0.3225071781007607, 0.7694015043666966,
      0.2184165276462826, 0.11296271530997704, 0.8297493455090494,
      0.16368411546542605, 0.12398551506473998, 0.3108305306090944,
      0.6336882049624735, 0.9880021890971149, 0.5243723210473457,
      0.05941932923237081, 0.795055539587106, 0.21827601798274454,
      0.7350970805982928, 0.9581390976142645, 0.024195987097207716,
      0.7857332501380803, 0.789034608246187, 0.273782286235279,
      0.06327743821384746, 0.8091213223648563, 0.12433072960297542,
      0.734536221709789, 0.9479018829139265, 0.24335759176696525,
      0.09584210273966365, 0.019417063444296945, 0.9050364474567667,
      0.39122027504982615, 0.8681712587555066, 0.12222345414409097,
      0.7354115831149068, 0.09426292551537618, 0.6562409417770831,
      0.9555096976744304, 0.22580604752401556, 0.09173880246535271,
      0.8467687324002768, 0.25331837654990363, 0.2971201930430514,
      0.6871771380023795, 0.4598490699465576, 0.8548997643095457,
      0.49716649211045927, 0.9629877000606215, 0.7454849738138569,
      0.6686006651562695, 0.23466958565807627, 0.5043864353303047,
      0.9280978816600116, 0.3619187102561381, 0.4943834318525533,
      0.8929168911408814, 0.5973577513175821, 0.9154977305523937,
      0.16167835696528932, 0.00879105994965479, 0.3722162788806449,
      0.4137110956387997, 0.7554796895827716, 0.4262947764620326,
      0.2821001878207565, 0.5851716626708432, 0.13036954246845434,
      0.6719286556216624, 0.46853279819678106, 0.8068739275865939,
      0.5246379669706098, 0.6696822511467659, 0.9851729651176886,
      0.6352529478531612, 0.9876483243370902, 0.4796096256740723,
      0.07828123220549243, 0.01666658117388553, 0.7713237801709629,
      0.19227539749714273, 0.6477442167804863, 0.8296108668873148,
      0.8408270637933359, 0.5858916746764142, 0.39207603358175436,
      0.6512116599302615, 0.22091722227321697, 0.26549468832118306,
      0.42463077223116685, 0.3739399243959183, 0.637220926668167,
      0.4127993086424804, 0.7330781748291435, 0.6846329557052264,
      0.8706283318141304, 0.8556744654927602, 0.026605726750204894,
      0.2156590125375315, 0.1720016142109777, 0.1636425493513094,
      0.2161311147363758, 0.6193051935313343, 0.1208148855721074,
      0.2147947208067733, 0.5477523376032087, 0.3244361774046971,
      0.776231896830816, 0.2898279277351228, 0.7048385408506992,
      0.5398424213945401, 0.7277954239753126, 0.7090784934805678,
      0.8168443286746514, 0.9655112715865066, 0.32879679748134083,
      0.0415373638495069, 0.27652193245783074, 0.5119631064448018,
      0.2003756717632863, 0.9547691662742952, 0.5532173266398412,
      0.7339211414038918, 0.541129796290132, 0.067753449194073,
      0.6895474397339463, 0.768028187222243, 0.5701898214986116,
      0.45984925367895224, 0.5848389202568094, 0.022061931735203766,
      0.4381188006175085, 0.1537714390160434, 0.6845104658444052,
      0.4787796674946396, 0.039245948023004296, 0.09305279297267677,
      0.5038779513568032, 0.5291200952347763, 0.47813922070253057,
      0.1763598696391846, 0.343344092674577, 0.43175412168739746,
      0.06537570518864877, 0.922608562765463, 0.3353584601247117,
      0.372108567649557, 0.15384704012707595, 0.46261835764392467,
      0.15814798764932503, 0.990793542803208, 0.03794525397537174,
      0.3351606884372109, 0.534840444366093, 0.23162411452633735,
      0.04367133628449538, 0.8316906562759616, 0.6566395125670361,
      0.37313069564228263, 0.9250233300605775, 0.898075024894391,
      0.7970161164108853, 0.801354872420249, 0.11309720476737684,
      0.8936412581599842, 0.6443397238714594, 0.9750419527602472,
      0.6280581940217098, 0.19637828090667742, 0.9939756900872305,
      0.89784898325502, 0.5041130309143562, 0.638165805274217,
      0.7280754559623015, 0.607985676544651, 0.6337092537378477,
      0.8122906640281615, 0.8861331413812303, 0.07791402882482235,
      0.12096761905638798, 0.7934301822018359, 0.6993190529455997,
      0.6946913943681419, 0.7220660122266722, 0.1326478910544331,
      0.9093438917314625, 0.7158973805974036, 0.0639233960876695,
      0.44570455933289743, 0.5007622104407956, 0.9822891994253189,
      0.5207628002794258, 0.7060340394261948, 0.1599905607784855,
      0.893795307965856, 0.8954830360688013, 0.5013712056251718,
      0.9346634792774808, 0.6995348762925893, 0.18730812688942167,
      0.029756141581942064, 0.7305713386373949, 0.17259795252106835,
      0.5613383169298747, 0.9813189919429297, 0.8868607988353276,
      0.780532464352305,
    ];
    jest.spyOn(global.Math, 'random').mockImplementation(() => {
      const res = randomValues.pop();
      if (typeof res === 'undefined') {
        throw new Error();
      }
      return res;
    });
  });

  afterEach(() => {
    jest.spyOn(global.Math, 'random').mockRestore();
  });
  it('snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ScramblePage />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();

    act(() => {
      jest.advanceTimersByTime(1000 / 60);
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
