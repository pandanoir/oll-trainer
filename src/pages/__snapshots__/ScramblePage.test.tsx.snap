// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScramblePage snapshot 1`] = `
<DocumentFragment>
  <div
    class="_StyledDiv-sc-1wlb5fx-0 fpjcVC"
  >
    <div
      class="_StyledDiv2-sc-1wlb5fx-1 flEBRo"
    >
      #1
    </div>
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events"
    >
      <div
        class="swiper-button-prev"
      />
      <div
        class="swiper-button-next"
      />
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ScramblePage snapshot 2`] = `
<DocumentFragment>
  <div
    class="_StyledDiv-sc-1wlb5fx-0 fpjcVC"
  >
    <div
      class="_StyledDiv2-sc-1wlb5fx-1 flEBRo"
    >
      #1
    </div>
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events"
    >
      <div
        class="swiper-button-prev"
      />
      <div
        class="swiper-button-next"
      />
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          F L' D' F R2 B' D' F2 R2 B' R D2 L' D L' D2 L B2 U' F
        </div>
        <div
          class="swiper-slide"
        >
          R' B2 D2 F R2 F2 U' R B2 D' L U2 B2 R2 B R' D' B' D' B'
        </div>
        <div
          class="swiper-slide"
        >
          B' U L U' B2 D2 R D' L' D L' F L2 U2 F2 D F' R U' B2
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
